lib/ansible/modules/network/dnac/dnac_rest.py

```diff
@@ -200,10 +201,10 @@ def main():
                                method=module.params['method'].upper(),
                                use_proxy=module.params['use_proxy'],
                                force=True,
-                               timeout=module.params['timeout']),
+                               timeout=module.params['timeout'],)
 
-    except Exception as e:
-        module.fail_json(msg=e.fp)
+    except Exception:
+        module.fail_json(msg=resp['status'])
```

Comment:
I am not sure this will work. It is unlikely that resp is set when there's an Exception.

Reply:
It's something I've struggled with. How do I catch HTTP exceptions since info and resp appear to not be created? Also, what exception should I catch since HTTPException and HTTPError aren't working for me?

Reply:
You have 2 cases, either the HTTP request fails (connection refused, host unreachable, ...) or the HTTP request worked. In the second case, the status could be 200, or something else. Normally fetch_url handle 3XX errors too. In this second case, resp will be returned and there's no exception.